tests/registry-e2e.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+use std::time::Duration;
+
+use chariott_common::proto::runtime::{
+    intent_registration::Intent, intent_service_registration::ExecutionLocality,
+};
+use common::get_uuid;
+use examples_common::chariott::{
+    api::{Chariott, GrpcChariott},
+    registration::Builder as RegistrationBuilder,
+};
+use tokio::time::*;
+
+mod common;
+
+#[tokio::test]
+async fn provider_is_registered_on_inspection() -> Result<(), anyhow::Error> {
+    // arrange
+    let namespace = format!("e2e.registration.{}", get_uuid());
+
+    let builder = RegistrationBuilder::new(
+        "e2e",
+        "1.0.0",
+        "http://localhost/".parse().unwrap(),
+        &namespace,
+        [Intent::Inspect],
+        ExecutionLocality::Local,
+    );
+
+    let mut chariott = setup().await;
+
+    // act
+    builder.register_once(&mut None, true).await?;
+
+    let initial_entries = chariott.inspect("system.registry", namespace.clone()).await?;
+    sleep(Duration::from_secs(5)).await;
```

Comment:
[nit] Do we want to add a comment around why we wait five seconds? With the context of the PR it's clear to me that we set the TTL to two seconds (though I think we also need to do this for the E2E tests in the CI), but it might be worth mentioning that information.

Reply:
I made this a bit more obvious and strict with a649b449c2c1e346bee77b2743b4d37661a04efc. Let me know what you think.